Use container log instead of termination log

gjkim42 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind cleanup

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:


Since the termination log cannot be accessed until the container is terminated, use the container log.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #
ref: https://github.com/kubernetes/kubernetes/pull/116398#issuecomment-1700995522

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

/cc @tzneal

gjkim42 committed:
Use container log instead of termination log

Since the termination log cannot be accessed until the container is
terminated, use the container log.

tzneal commented on test/e2e_node/container_lifecycle_pod_construction.go:
```diff
@@ -250,21 +252,28 @@ func (o containerOutputList) findIndex(name string, command string, startIdx int
+		log, err := e2epod.GetPodLogs(ctx, f.ClientSet, f.Namespace.Name, pod.Name, cs.Name)
+		framework.ExpectNoError(err)
+		buf.WriteString(log)
+		//		// If the container is terminated but the reason is ContainerStatusUnknown,
```
Remove before merge, assuming you left it in temporarily just to ensure all tests pass.

gjkim42 commented:
/retest

gjkim42 commented:
/test pull-kubernetes-node-e2e-containerd-sidecar-containers
/test pull-kubernetes-e2e-gce-cos-alpha-features

gjkim42 commented:
/retest

gjkim42 commented:
/retest

gjkim42 commented:
/test pull-kubernetes-node-e2e-containerd-sidecar-containers
/test pull-kubernetes-e2e-gce-cos-alpha-features

gjkim42 commented:
/retest

gjkim42 commented:
/cc @tzneal 
Could you take a look?

/assign @SergeyKanzhelev

tzneal reviewed (approved):
/lgtm

gjkim42 commented:
this is needed to test the https://github.com/kubernetes/kubernetes/pull/120269.

SergeyKanzhelev reviewed (approved):
/lgtm
/approve

gjkim42 commented:
/hold 
in case there comes new usages for parseLog.
I'll rebase the upstream later and merge it.

gjkim42 commented:
/test pull-kubernetes-node-e2e-containerd-sidecar-containers

gjkim42 commented:
> /hold
> in case there comes new usages for parseLog.
> I'll rebase the upstream later and merge it.

/unhold

rebased the upstream and fixed the new usage.

Could you PTAL? @SergeyKanzhelev @tzneal

gjkim42 commented:
/retest

tzneal commented:
/lgtm
